Modules/gcmodule.c

```diff
@@ -1405,6 +1411,36 @@ gc_is_tracked(PyObject *module, PyObject *obj)
     return result;
 }
 
+PyDoc_STRVAR(gc_freeze__doc__,
+"freeze() -> None\n"
+"\n"
+"Freeze all current tracked objects and ignore them for future collections.\n"
+"This can be used before a fork to make the gc copy-on-write friendly.\n"
+"Note: collection before a fork may free pages for future allocation\n"
+"which can cause copy-on-write.\n"
+);
+
+static PyObject *
+gc_freeze(PyObject *module)
+{
+    for (int i = 0; i < NUM_GENERATIONS; ++i) {
+        gc_list_merge(GEN_HEAD(i), &_PyRuntime.gc.permanent_generation.head);
+        _PyRuntime.gc.generations[i].count = 0;
+    }
+    Py_RETURN_NONE;
+}
+
+PyDoc_STRVAR(gc_get_freeze_stats__doc__,
+"get_freeze_stats() -> n\n"
```

Comment:
Since this is not equivalent to what `get_stats()` is returning, can you rename to `get_frozen_count()`?